src/components/Navigator/QuickNavigationModal.vue

```diff
@@ -0,0 +1,236 @@
+          regexPattern += '|';
+        }
+        regexPattern += `(?<!${char}`;
+        if (idx !== 0 && char === [...debouncedInput][idx - 1]) {
+          regexPattern += char;
+        }
+        regexPattern += '[^';
+        if (idx !== 0) {
+          regexPattern += [...debouncedInput][idx - 1];
+        }
+        regexPattern += char;
+        regexPattern += `]*?)${char}`;
+      });
+      return new RegExp(regexPattern, 'gi');
+    },
+  },
+  methods: {
+    closeQuickNavigationModal() {
+      this.quickNavigationStore.toggleShowQuickNavigationModal();
+    },
+    constructFuzzyRegex(userInput) {
+      // Construct regex for fuzzy match
+      // Ex:
+      // foobar -> f[^f]*?o[^o]*?o[^o]*?b[^b]*?a[^a]*?r
+      let regexBuilder = '';
+      [...userInput].forEach((char, idx) => {
+        regexBuilder += char.toLowerCase();
+        if (idx !== userInput.length - 1) {
+          regexBuilder += `[^${char.toLowerCase()}]*?`;
+        }
+      });
+      return regexBuilder;
+    },
+    debounceInput: debounce(function debounceInput(value) {
+      this.debouncedInput = value;
+    }, 500),
+    fuzzyMatch({ debouncedInput, symbols, processedInputRegex }) {
+      const matches = [];
+      symbols.map((symbol) => {
+        if (debouncedInput) {
+          const match = processedInputRegex.exec(symbol.title.toLowerCase());
+          if (match) {
+            const isExactMatch = RegExp(
+              debouncedInput,
+            ).test(symbol.title.toLowerCase());
+            if (Math.abs(match[0].length - debouncedInput.length) > debouncedInput.length * 2) {
+              return false;
+            }
+            matches.push({
+              title: symbol.title,
+              path: symbol.path,
+              inputLengthDifference: Math.abs(
+                symbol.title.length - debouncedInput.length,
+              ),
+              matchLength: match[0].length,
+              matchLengthDifference: isExactMatch === true
+                ? 0
+                : Math.abs(match[0].length - debouncedInput.length),
+              start: match.index,
+              substring: match[0],
+            });
+            return true;
+          }
+        }
+        return false;
+      });
+      return matches;
+    },
+    orderSymbolsByPriority(matchingSymbols) {
+      return matchingSymbols.sort((a, b) => {
+        // Sort exact string match by symbol title length
+        if (a.matchLengthDifference === 0 && b.matchLengthDifference === 0) {
+          if (a.inputLengthDifference > b.inputLengthDifference) return 1;
+          if (a.inputLengthDifference < b.inputLengthDifference) return -1;
+        }
+        // Exact string match have priority over fuzzy match
+        if (a.matchLengthDifference === 0 || b.matchLengthDifference === 0) {
+          if (a.matchLengthDifference > b.matchLengthDifference) return 1;
+          if (a.matchLengthDifference < b.matchLengthDifference) return -1;
+        }
```

Comment:
In case any of the matchLengthDifference are 0 you will be ordering in line 187 anyways. We could delete this.